engine/hooks/bitbucket_server.go

```diff
@@ -0,0 +1,349 @@
+package hooks
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/ovh/cds/sdk"
+	"github.com/ovh/cds/sdk/log"
+	"strings"
+)
+
+func (s *Service) generatePayloadFromBitbucketServerRequest(t *sdk.TaskExecution, event string) ([]map[string]interface{}, error) {
+	payloads := []map[string]interface{}{}
+	switch event {
+	case "repo:refs_changed":
+		return s.generatePayloadFromBitbucketServerPushEvent(t)
+	case "pr:opened":
+		return s.generatePayloadFromBitbucketServerPROpened(t)
+	case "pr:modified":
+		return s.generatePayloadFromBitbucketServerPRModified(t)
+	case "pr:declined":
+		return s.generatePayloadFromBitbucketServerPRDeclined(t)
+	case "pr:deleted":
+		return s.generatePayloadFromBitbucketServerPRDeleted(t)
+	case "pr:merged":
+		return s.generatePayloadFromBitbucketServerPRMerged(t)
+	case "pr:comment:added":
+		return s.generatePayloadFromBitbucketServerPRCommentAdded(t)
+	case "pr:comment:edited":
+		return s.generatePayloadFromBitbucketServerPRCommentEdited(t)
+	case "pr:comment:deleted":
+		return s.generatePayloadFromBitbucketServerPRCommentDeleted(t)
+	case "pr:reviewer:approved":
+		return s.generatePayloadFromBitbucketServerPRReviewerApproved(t)
+	case "pr:reviewer:updated":
+		return s.generatePayloadFromBitbucketServerPRReviewerUpdated(t)
+	case "pr:reviewer:unapproved":
+		return s.generatePayloadFromBitbucketServerPRReviewerUnapproved(t)
+	case "pr:reviewer:needs_work":
+		return s.generatePayloadFromBitbucketServerPRReviewerNeedsWork(t)
+	default:
```

Comment:
should we have a default behavior on unmanaged event type or should we doesn't care about them ?